### PR TITLE
Check geolocation quota at trip end

### PR DESCRIPTION
**Pull request by zatteo**
What do you think about this event implementation @Ldoppea and @paultranvan.

It means that we must install "events" package on CozyGPSMemory but that's all.

### CODE REVIEW

File: src/app/domain/geolocation/tracking/events.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { EventEmitter } from 'events'
+
+export const GeolocationTrackingEmitter = new EventEmitter()
+
+// Available events
+export const TRIP_END = 'TRIP_END'
```
Comment (paultranvan) [inline src/app/domain/geolocation/tracking/events.js:6]: Would you mind moving this to the "tracking/consts.js" file ? :)

File: src/app/domain/geolocation/hooks/tracking.ts [hunk 2]
```diff
@@ -1,16 +1,38 @@
...
 
 export const useGeolocationTracking = (): void => {
   const client = useClient()
 
+  useEffect(() => {
+    if (!client) return
+
+    const onTripEnd = (): void => {
+      log.debug('Trip end event received, checking quota')
+      void checkGeolocationQuota(client)
```
Comment (paultranvan) [inline src/app/domain/geolocation/hooks/tracking.ts:26]: That's the first time I see an explicit `void` in a function call. Is it required by typescript?

**Review by paultranvan — APPROVED** [review]
Some small remarks, but looks good overall

File: src/app/domain/geolocation/helpers/quota.ts [hunk 3]
```diff
@@ -43,18 +42,14 @@ const getFirstTimeserie = async (
...
           .indexFields(['startDate'])
           .sortBy([{ startDate: 'asc' }])
-          .limitBy(1),
-        options: {
-          as: 'io.cozy.timeseries.geojson/firstTimeserie',
-          fetchPolicy: CozyClient.fetchPolicies.olderThan(ONE_DAY)
-        }
-      })) as unknown as { data: FirstTimeserie[] }
+          .limitBy(1)
+      )) as unknown as { data: FirstTimeserie[] }
```
[the remark this answers is not in the document]
Comment (zatteo) [reply src/app/domain/geolocation/helpers/quota.ts:52]: Yes I tried to explain it in the commit message. 

Here we handle cache manually in AsyncStorage. So we always want to do the request here.

[the remark this answers is not in the document]
Comment (zatteo) [reply src/app/domain/geolocation/helpers/quota.ts:52]: More detailed explanation: we want to do the request until we have a timeserie. As soon as we have a timeserie, we do not need anymore to do this request.

I think this behavior is not possible to model in cozy-client.

So I do the request as long as the result is [], and then when there is a timeserie, I store the timeserie in AsyncStorage. And if there is a timeserie in AsyncStorage, I don't do the request.

**About this commit**

With the cozy-client store and 24 hours as cache as before this commit, I could store [] in the cache which is NOT what I want.

[the remark this answers is not in the document]
Comment (paultranvan) [reply src/app/domain/geolocation/helpers/quota.ts:52]: Ok, that's much clearer, thanks!

**Pull request merged**